x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/assignment/ChunkingSettingsFeatureFlag.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.core.ml.inference.assignment;
```

Comment:
Good catch. I was following the example with the AdaptiveAllocationFeatureFlag and must misunderstood the purpose of the `assignment` directory. I'll move this file to the inference plugin.

Reply:
I tried moving this into the inference plugin but what I'm finding is that if we want to reuse this feature flag for the inference processor work we hope to do in the future, we run into issues importing from the `org.elasticsearch.xpack.core.inference` package as it creates a circular reference. I'm wondering if it makes more sense to move this to `org.elasticsearch.xpack.core.inference` to avoid this but I'm not sure we have a great place for it. It could go in [this directory](https://github.com/elastic/elasticsearch/tree/main/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/inference) but let me know if you have a suggestion for a better place for this?